Fix window is resizable in legacy full screen mode, #5609

low-batt opened the pull request:
This commit will:
- Change `legacyAnimateToFullscreen` to remove `resizable` from the window `styleMask`
- Change `legacyAnimateToWindowed` to insert `resizable` into the `styleMask`
- Change `legacyAnimateToWindowed` to use `disableAnimations` instead of `legacyFullScreenAnimation`
- Remove `legacyFullScreenAnimation` from `Preferences`
- Change the observer for `didChangeScreenParametersNotification` to not call `setWindowFrameForLegacyFullScreen` if the screen count did not change

The change to remove `resizable` from the window `styleMask` will prevent the window from being resized while in full screen mode.

This commit will also replace the `legacyFullScreenAnimation` setting for which there was no UI with the new `disableAnimations` setting which can be set using IINA's settings panel.

When entering/exiting full screen mode IINA changes the app presentation options to enable/disable hiding of the menu bar and dock. As the menu bar and dock are sliding away the observer for
`didChangeScreenParametersNotification` because the moving of the menu bar and dock result in changes to the dimensions of the visible frame. The change to only call `setWindowFrameForLegacyFullScreen` when the screen count changes eliminates extra drawing that disrupts the full screen transition animations.

- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #5609.

---

**Description:**

low-batt committed:
Fix window is resizable in legacy full screen mode, #5609

This commit will:
- Change legacyAnimateToFullscreen to remove resizable from the window
  styleMask
- Change legacyAnimateToWindowed to insert resizable into the styleMask
- Change legacyAnimateToWindowed to use disableAnimations instead of
  legacyFullScreenAnimation
- Remove legacyFullScreenAnimation from Preferences
- Change the observer for didChangeScreenParametersNotification to not
  call setWindowFrameForLegacyFullScreen if the screen count did not
  change

The change to remove resizable from the window styleMask will prevent
the window from being resized while in full screen mode.

This commit will also replace the legacyFullScreenAnimation setting for
which there was no UI with the new disableAnimations setting which can
be set using IINA's settings panel.

When entering/exiting full screen mode IINA changes the app presentation
options to enable/disable hiding of the menu bar and dock. As the menu
bar and dock are sliding away the observer for
didChangeScreenParametersNotification because the moving of the menu bar
and dock result in changes to the dimensions of the visible frame. The
change to only call setWindowFrameForLegacyFullScreen when the screen
count changes eliminates extra drawing that disrupts the full screen
transition animations.

uiryuu commented:
Do you also want to disable the animation for legacy fs for Tahoe in this PR? The legacy fs animation is still laggy on this commit.

low-batt committed:
Fix window is resizable in legacy full screen mode, #5609

This commit will suppress enter/exit animation for legacy full screen
mode if running under macOS 26 to address lagging experienced with the
macOS Tahoe beta.

low-batt commented:
In the latest commit I've disabled animations for legacy full screen under macOS Tahoe for now.
